src/read.rs

```diff
@@ -645,6 +645,39 @@ impl<R: Read + Seek> ZipArchive<R> {
         Ok(shared)
     }
 
+    /// Returns the verification value and salt for the AES encryption of the file
+    ///
+    /// It fails if the file is not encrypted or if the file number is invalid.
+    ///
+    /// # Returns
+    ///
+    /// - Some with the verification value and the salt
+    /// - None if the file is not encrypted with AES
+    #[cfg(feature = "aes-crypto")]
+    pub fn get_aes_verification_key_and_salt(
+        &mut self,
+        file_number: usize,
+    ) -> ZipResult<Option<(AesMode, Vec<u8>, Vec<u8>)>> {
+        let (_, data) = self
+            .shared
+            .files
+            .get_index(file_number)
+            .ok_or(ZipError::FileNotFound)?;
+
+        if !data.encrypted {
+            return Err(ZipError::UnsupportedArchive(ZipError::ARCHIVE_NOT_ENCRYPTED));
```

Comment:
Shouldn't the return value be `Ok(None)` in this case? If not, what's the difference?

Reply:
`Err` means the archive is not encrypted whereas `Ok(None)` means the archive is encrypted but not with AES.

I will remove the distinction by not checking whether it is encrypted to simplify things :)